src/libsyntax/parse/parser.rs

```diff
@@ -5809,13 +5811,19 @@ impl<'a> Parser<'a> {
                                     maybe_append(attrs, extra_attrs));
             return Ok(Some(item));
         }
-        if self.check_keyword(keywords::Unsafe) &&
-            self.look_ahead(1, |t| t.is_keyword(keywords::Impl))
+        if (self.check_keyword(keywords::Unsafe) &&
```

Comment:
Okay, I guess 2 look aheads is fine for now, although it is pretty obvious this needs some refactor, which I’ll be doing later anyway.